apigee_edge.module

```diff
@@ -1121,6 +1121,11 @@ function template_preprocess_app_credential(array &$variables) {
  * it has been in Apigee Edge.
  */
 function apigee_edge_user_presave(UserInterface $account) {
+  // Do not sync the anonymous user.
+  if ($account->isAnonymous()) {
```

Comment:
This should not happen... nobody should update/delete the definition of the anonymous user. This only happens because you are doing something that you should not in  `_apigee_edge_rename_anonymous_user()`.

Reply:
There are situations other than ` _apigee_edge_rename_anonymous_user` where this could happen. There's a check, which you added, for the anonymous user in `apigee_edge_user_delete` (https://github.com/apigee/apigee-edge-drupal/commit/9de5c40b58730dd5ceb8e6afb47ac0d54aca3a70) so I don't see why there should not be one during presave.

Reply:
Okay, as an extra sanity check you can also make sure that someone did not call mistakenly the save() on the anonymous user, but the sync already excludes the anonymous user from the synching this is the reason why we did not need this extra sanity check before, only now, when you added `_apigee_edge_rename_anonymous_user()`.

This condition also excludes the anonymous user: https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/src/Job/DeveloperSync.php#L85

Reply:
Not having an extra condition here is actually useful, because the actual requirement is not being a logged-in user, rather having all required fields (email, first name, last name, username) by Apigee Edge filled in when someone/something saves a user. If you have ever integrated SAML with the Apigee Edge module you know that most SAML modules tries to create an empty stub user first with invalid, missing data, usually with a fake email addresses, that could lead to possible errors in the future if those stub users gets saved to Apigee Edge. The current implements prevents this.